### PR TITLE
OSDOCS-1861: Add discussion of multi-network policy

### PR DESCRIPTION
**Pull request by jboxman**
- https://issues.redhat.com/browse/OSDOCS-1861

~So this rebased off of https://github.com/openshift/openshift-docs/pull/32708 and all those changes appear here as well. Ideally, 32708 is merged first.~ The following content is specific to this PR:

Previews
- [Configuring multi-network policy](https://deploy-preview-32252--osdocs.netlify.app/openshift-enterprise/latest/networking/multiple_networks/configuring-multi-network-policy.html)

(So mostly `modules/nw-multi-network-policy-*.adoc` files.)

### CODE REVIEW

**Comment by jboxman** [comment]
Hi @s1061123, can you take a look at this multi-network policy content when you have a moment? Thanks!

File: modules/nw-networkpolicy-create.adoc [hunk 1]
```diff
@@ -1,59 +1,120 @@
...
+endif::multi[]
+ifndef::multi[]
 +
-[source,terminal]
+.Allow ingress from all pods in the same namespace
+[source,yaml]
 ----
-$ oc create -f <policy-name>.yaml -n <project>
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
```
Comment (weliang1) [inline modules/nw-networkpolicy-create.adoc:83]: Should be apiVersion: k8s.cni.cncf.io/v1beta1?

File: modules/nw-networkpolicy-create.adoc [hunk 2]
```diff
@@ -1,59 +1,120 @@
...
+--
+
+.. Define a policy object in the file that you just created, such as in the following examples:
++
+.Deny ingress from all pods in all namespaces
 [source,yaml]
 ----
+ifndef::multi[]
 kind: NetworkPolicy
 apiVersion: networking.k8s.io/v1
```
Comment (weliang1) [inline modules/nw-networkpolicy-create.adoc:53]: Should it be apiVersion: k8s.cni.cncf.io/v1beta1?

File: modules/nw-networkpolicy-delete.adoc [hunk 3]
```diff
@@ -1,25 +1,41 @@
...
 
 .Procedure
 
-* To delete a `NetworkPolicy` object, enter the following command. Replace `<policy-name>` with the name of the object.
+* To delete a `NetworkPolicy` object, enter the following command:
 +
 [source,terminal]
 ----
-$ oc delete networkpolicy <policy-name>
+$ oc delete networkpolicy <policy_name>
```
Comment (weliang1) [inline modules/nw-networkpolicy-delete.adoc:28]: oc delete multi-networkpolicy

File: modules/nw-networkpolicy-delete.adoc [hunk 4]
```diff
@@ -1,25 +1,41 @@
...
+--
+where:
+
+`<policy_name>`:: Specifies the name of the network policy.
+--
++
+.Example output
+[source,text]
+----
+networkpolicy.networking.k8s.io/allow-same-namespace deleted
```
Comment (weliang1) [inline modules/nw-networkpolicy-delete.adoc:40]: Change to "multinetworkpolicy.k8s.cni.cncf.io "ingress-allow-diff-podselector-with-same-namespaceselector" deleted
"

File: modules/nw-networkpolicy-create.adoc (hunk 2 above)
Comment (jboxman) [reply modules/nw-networkpolicy-create.adoc:65]: @weliang1, yes, I've added a conditional for it because this file is included in 2 different assemblies.

**Comment by jboxman** [comment]
@weliang1, I've linked to additional resources for delete, edit, ect. I expect that a user understands that these require using `multi-networkpolicy` instead of `networkpolicy` in commands.

```
* xref:../../networking/network_policy/about-network-policy.adoc#about-network-policy[About network policy]
* xref:../../networking/network_policy/viewing-network-policy.adoc#viewing-network-policy[Viewing a network policy]
* xref:../../networking/network_policy/editing-network-policy.adoc#editing-network-policy[Editing a network policy]
* xref:../../networking/network_policy/deleting-network-policy.adoc#deleting-network-policy[Deleting a network policy]
```

Although I can try to make it clear in the introduction to multi-network policy.

File: modules/nw-networkpolicy-delete.adoc (hunk 4 above)
Comment (jboxman) [reply modules/nw-networkpolicy-delete.adoc]: @weliang1, would this be correct for creating a multi-network policy?

`multinetworkpolicy.k8s.cni.cncf.io/default-deny created`

File: modules/nw-multi-network-policy-enable.adoc [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+As a cluster administrator, you can enable multi-network policy support on your cluster.
+
+.Prerequisites
+
+* Install the OpenShift CLI (`oc`).
+* Log in to the cluster with a user with `cluster-admin` privileges.
+
+.Procedure
+
+. Create a file with the following YAML:
```
Comment (mikemckiernan) [inline modules/nw-multi-network-policy-enable.adoc:17]: Nothing wrong with what you have, but I'll share an idea.  You could be prescriptive and suggest creating a file named `multi-network-policy.yaml` in this step and then reference the same file name in the next.  That would eliminate the somewhat superfluous suggestion to substitute the real file name.  I imagine that customers understand they can name files whatever they want.

File: modules/nw-multi-network-policy-enable.adoc [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+----
+apiVersion: operator.openshift.io/v1
+kind: Network
+metadata:
+  name: cluster
+spec:
+  useMultiNetworkPolicy: true
+----
+
+. To enable multi-network policy, enter the following command. Replace `<name>` with the name of the YAML file you created in the previous step.
```
Comment (mikemckiernan) [inline modules/nw-multi-network-policy-enable.adoc:29]: Maybe "Configure the cluster to enable multi-network policy:"

I believe you stated a preference for "To <goal>, enter..." but that ends up with sentence structure that isn't parallel with the first step.  I also suspect that you can drop the "Replace..." sentence even if you don't choose to be prescriptive.

File: modules/nw-networkpolicy-create.adoc [hunk 7]
```diff
@@ -6,6 +6,11 @@
 ifeval::[{product-version} >= 4.6]
 :ovn:
 endif::[]
+ifeval::["{context}" == "configuring-multi-network-policy"]
+:multi:
+endif::[]
+
+[id="nw-networkpolicy-create_{context}"]
 
 [id="nw-networkpolicy-create_{context}"]
```
Comment (mikemckiernan) [inline modules/nw-networkpolicy-create.adoc:15]: Just an eye check...did you mean to add this again?

**Review by mikemckiernan — APPROVED** [review]
Looks good.

**Comment by weliang1** [comment]
Not see "oc get multi-networkpolicy" in https://deploy-preview-32252--osdocs.netlify.app/openshift-enterprise/latest/networking/network_policy/viewing-network-policy.html#viewing-network-policy
Not see "oc edit multi-networkpolicy" in https://deploy-preview-32252--osdocs.netlify.app/openshift-enterprise/latest/networking/network_policy/editing-network-policy.html#editing-network-policy
Not see "oc delete multi-networkpolicy" in https://deploy-preview-32252--osdocs.netlify.app/openshift-enterprise/latest/networking/network_policy/deleting-network-policy.html#deleting-network-policy

**Comment by weliang1** [comment]
> @weliang1 multi-networkpolicy and Kuberneres networkpolicy are different feature, so I believe we should have different pages for each feature, otherwise user may get confused about multi-networkpolicy and Kubernetes networkpolicy.

@jboxman I agree @s1061123 suggestion.

**Comment by jboxman** [comment]
@weliang1 @s1061123 

I've included all the multi-network policy procedures. I use conditional expressions to use `multi-networkpolicy` in place of `networkpolicy`.

https://deploy-preview-32252--osdocs.netlify.app/openshift-enterprise/latest/networking/multiple_networks/configuring-multi-network-policy.html

**Comment by weliang1** [comment]
> @weliang1 @s1061123
> 
> I've included all the multi-network policy procedures. I use conditional expressions to use `multi-networkpolicy` in place of `networkpolicy`.
> 
> https://deploy-preview-32252--osdocs.netlify.app/openshift-enterprise/latest/networking/multiple_networks/configuring-multi-network-policy.html

@jboxman should you include "oc edit multi-networkpolicy" and "oc delete multi-networkpolicy" in https://deploy-preview-32252--osdocs.netlify.app/openshift-enterprise/latest/networking/multiple_networks/configuring-multi-network-policy.html?

**Comment by jboxman** [comment]
@weliang1 thank you!

@mikemckiernan, I changed the structure some and added conditionals everywhere, can you take a look when you have a minute? Thanks!

**Pull request merged**